snuba/query/data_source/simple.py

```diff
@@ -0,0 +1,18 @@
+from dataclasses import dataclass
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.datasets.entities import EntityKey
+from snuba.query.data_source import DataSource
+
+
+@dataclass(frozen=True)
+class Entity(DataSource):
+    """
+    Represents an Entity in the logical query.
+    """
+
+    key: EntityKey
+    schema: ColumnSet
```

Comment:
At what point will we expand the Entity data model to include join relationships?

Reply:
I'd say after we are capable to have a nested structure in the queries, but there is no strict dependency on the order of these two steps. Did not start with that because there is still quite a bit of work going on on the entities definition and I wanted to minimize the overlap.